src/category_theory/abelian/exact.lean

```diff
@@ -343,6 +348,91 @@ begin
   rw [fork.ι_of_ι, cofork.π_of_π, ← L.map_comp, kernel_comp_cokernel _ _ e1, L.map_zero]
 end
 
+end
+
+section
+
+variables (h : ∀ {X Y Z : A} {f : X ⟶ Y} {g : Y ⟶ Z}, exact f g → exact (L.map f) (L.map g))
+include h
+
+open_locale zero_object
+
+/-- A functor which preserves exactness preserves zero morphisms. -/
+lemma preserves_zero_morphisms_of_map_exact : L.preserves_zero_morphisms :=
+begin
+  replace h := (h (exact_of_zero (𝟙 0) (𝟙 0))).w,
+  rw [L.map_id, category.comp_id] at h,
+  exact preserves_zero_morphisms_of_map_zero_object (id_zero_equiv_iso_zero _ h),
+end
+
+/-- A functor which preserves exactness preserves monomorphisms. -/
+lemma preserves_monomorphisms_of_map_exact : L.preserves_monomorphisms :=
+{ preserves := λ X Y f hf,
+  begin
+    letI := preserves_zero_morphisms_of_map_exact L @h,
+    apply ((tfae_mono (L.obj 0) (L.map f)).out 2 0).mp,
+    rw ←L.map_zero,
+    exact h (((tfae_mono 0 f).out 0 2).mp hf)
+  end }
+
+/-- A functor which preserves exactness preserves epimorphisms. -/
+lemma preserves_epimorphisms_of_map_exact : L.preserves_epimorphisms :=
+{ preserves := λ X Y f hf,
+  begin
+    letI := preserves_zero_morphisms_of_map_exact L @h,
+    apply ((tfae_epi (L.obj 0) (L.map f)).out 2 0).mp,
+    rw ←L.map_zero,
+    exact h (((tfae_epi 0 f).out 0 2).mp hf)
+  end }
+
+/-- A functor which preserves exactness preserves kernels. -/
+def preserves_kernels_of_map_exact (X Y : A) (f : X ⟶ Y) :
+  preserves_limit (parallel_pair f 0) L :=
+{ preserves := λ c ic,
+  begin
+    letI := preserves_zero_morphisms_of_map_exact L @h,
+    letI := preserves_monomorphisms_of_map_exact L @h,
+    letI := mono_of_is_limit_fork ic,
+    have hf := (is_limit_map_cone_fork_equiv' L (kernel_fork.condition c)).symm
+      (is_limit_of_exact_of_mono (L.map (fork.ι c)) (L.map f)
+        (h (exact_of_is_kernel (fork.ι c) f (kernel_fork.condition c)
+          (ic.of_iso_limit (iso_of_ι _))))),
+    exact hf.of_iso_limit ((cones.functoriality _ L).map_iso (iso_of_ι _).symm),
+  end }
+
+/-- A functor which preserves exactness preserves zero cokernels. -/
+def preserves_cokernels_of_map_exact (X Y : A) (f : X ⟶ Y) :
+  preserves_colimit (parallel_pair f 0) L :=
+{ preserves := λ c ic,
+  begin
+    letI := preserves_zero_morphisms_of_map_exact L @h,
+    letI := preserves_epimorphisms_of_map_exact L @h,
+    letI := epi_of_is_colimit_cofork ic,
+    have hf := (is_colimit_map_cocone_cofork_equiv' L (cokernel_cofork.condition c)).symm
+      (is_colimit_of_exact_of_epi (L.map f) (L.map (cofork.π c))
+        (h (exact_of_is_cokernel f (cofork.π c) (cokernel_cofork.condition c)
+          (ic.of_iso_colimit (iso_of_π _))))),
+    exact hf.of_iso_colimit ((cocones.functoriality _ L).map_iso (iso_of_π _).symm),
+  end }
+
+/-- A functor which preserves exactness is left exact, i.e. preserves finite limits. -/
```

Comment:
Please reference the converse here, and reference this result in the docstring of the converse.